test_util/marathon.py

```diff
@@ -202,6 +202,85 @@ def _pool_for_marathon_app(app_id):
             raise Exception("Application deployment failed - operation was not "
                             "completed in {} seconds.".format(timeout))
 
+    def deploy_test_pod_and_check(self, pod, test_uuid):
+        with deploy_pod_and_cleanup(pod) as instances:
+            assert pod['scaling']['instances'] == instances
```

Comment:
I would not include this deploy_test_pod_and_check method as this is only asserting on instance count. Rather, we should block on instance count. The intention behind deploy_test_add_and_check is that there is a well known app that tests a variety of features and we want that variety checked whereas the pod for this case is truly a smoke test and trivial to check